app/models/batch_process.rb

```diff
@@ -95,22 +95,16 @@ def create_parent_objects_from_oids(oids, metadata_sources, adminset_keys)
         batch_processing_event("Skipping row [#{index}] with unknown admin set [#{adminset_key}] for parent: #{oid}", 'Skipped Row')
         next
       end
+      next unless user_create_permission(index, admin_set, oid)
+
       po = ParentObject.where(oid: oid).first_or_create do |parent_object|
         # Only runs on newly created parent objects
         setup_for_background_jobs(parent_object, metadata_source)
         parent_object.admin_set = admin_set
         fresh = true
-        unless current_ability.can?(:create, parent_object)
-          batch_processing_event("Skipping row [#{index}] because of access violation for parent: #{oid}", 'Skipped Row')
-          parent_object.destroy
```

Comment:
we removed this because we were getting a `frozen hash` error, which we believe is related to solr trying to delete the record after it was removed from the db.
https://www.ruby-forum.com/t/cant-modify-frozen-hash-during-validation/147127/4